### PR TITLE
Add support for displaying most used language (or 2nd, 3rd, 4th, ...)

### LINKED ISSUE #87: Primary language badge
I think it would be nice to have a badge for the language with the most lines (probably by LoC)? It could look something like this: Primary Language Badge

It would be cool if each language had its own color, but to get the feature started it might be best if all languages used the same color (blue) as the other badges.

### PR DESCRIPTION
**Pull request by elliotwutingfeng**
Closes #87

Continues where #96 left off. Thank you @spenserblack for starting this.

Display the **n**-th most used language by enabling `?showLanguage=true` and using the  `?languageRank=n` query string, where **n** ∈[1, number of languages detected by tokei]

For example,

```markdown
https://tokei-rs.onrender.com/b1/github/XAMPPRocky/tokei?showLanguage=true&languageRank=1&label=Champion&style=for-the-badge)](https://github.com/XAMPPRocky/tokei
```

## Edge cases

- If `?showLanguage=true` is not present, line count will be displayed as per default badge behavior.
- If  `?showLanguage=true` but  `?languageRank=` not present, badge will display most used language.
- If **n** ∉[1, number of languages detected by tokei], badge will display "N/A".
- If no languages are detected by tokei, badge will display "No Language".

## Live Demo

Repo: <https://github.com/XAMPPRocky/tokei>

[Champion](https://github.com/XAMPPRocky/tokei)

[First Runner Up](https://github.com/XAMPPRocky/tokei)

[Third Place](https://github.com/XAMPPRocky/tokei)

[languageRank0](https://github.com/XAMPPRocky/tokei)

[languageRank 100](https://github.com/XAMPPRocky/tokei)

[languageRank 10000000000000000000000000000000000000000](https://github.com/XAMPPRocky/tokei)

[No Languages](https://github.com/XAMPPRocky/tokei)

Colors, badge styles, logos, and labels can be freely customized by the user.

---

It is possible to enable negative integers and get least-used, 2nd least used languages, ..., but this would probably be out of scope here.

~Is `ranking` a suitable query parameter name?~

### CODE REVIEW

**Comment by XAMPPRocky** [comment]
Thank you for your PR! Sorry for the delay, but now that I merged your other PR, this has conflicts. Would you be interested in updating this one?

File: Earthfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-VERSION 0.7
+VERSION 0.6
```
Comment (XAMPPRocky) [inline Earthfile:1]: This shouldn't downgrade.

File: src/main.rs [hunk 2]
```diff
@@ -91,6 +92,8 @@ struct BadgeQuery {
     color: Option<String>,
     logo: Option<String>,
     r#type: Option<String>,
+    showLanguage: Option<String>,
+    languageRank: Option<String>,
```
Comment (XAMPPRocky) [inline src/main.rs:96]: The casing here is wrong, it should snake_case.

File: Earthfile (hunk 1 above)
Comment (elliotwutingfeng) [reply Earthfile:1]: I wasn't able to get it to work on onrender.com, I'll try 0.7 again.

File: src/main.rs [hunk 3]
```diff
@@ -216,17 +228,34 @@ async fn create_badge(
...
+    } else {
+        entry
+            .value
+            .into_iter()
+            .filter(|(language_type, _)| language_types.contains(&language_type))
+            .into_iter()
+            .collect()
+    };
+    let ranking_language = if !show_language {
+        "".to_owned()
```
Comment (XAMPPRocky) [inline src/main.rs:242]:
```suggestion
        String::new()
```

**Comment by elliotwutingfeng** [comment]
Haven't looked into this for a while, will try deploying on onrender.com

File: src/main.rs (hunk 2 above)
Comment (elliotwutingfeng) [reply src/main.rs:96]: Given that shields.io query parameters usually use camel case (e.g. https://shields.io/badges/lines-of-code), would it be better to use camel case here?

File: Earthfile (hunk 1 above)
Comment (elliotwutingfeng) [reply Earthfile:1]: GitHub CI still doesn't push the image to ghcr when on Earthly v0.7. v0.6 still works fine.

File: src/main.rs (hunk 2 above)
Comment (XAMPPRocky) [reply src/main.rs:96]: Sure, but the Rust code should still be in snake case, and you should use `#[serde(rename)]` for that.

File: Earthfile (hunk 1 above)
Comment (XAMPPRocky) [reply Earthfile:1]: What do you mean by "doesn't push", is there an error?

Comment (elliotwutingfeng) [reply Earthfile:1]: v0.7 (doesn't push to ghcr)

https://github.com/elliotwutingfeng/tokei_rs/actions/runs/5494842291/jobs/10013869341

Under `Run build` step

```code
              context | --> local context .
              output | --> exporting outputs

 Push Summary ⏫
————————————————————————————————————————————————————————————————————————————————

Did not push image /cache/build
Did not push image /cache/deps
Pushed image github.com/elliotwutingfeng/tokei_rs:dev+docker as 
Did not push image /cache/install-chef
Did not push image /cache/prepare

 Local Output Summary 🎁 (disabled)
```


v0.6 (still works)

https://github.com/elliotwutingfeng/tokei_rs/actions/runs/5494972413/jobs/10014066312

Under `Run build` step

```code
              context | --> local context .
              output | --> exporting outputs
              output | [          ]   0% pushing layers
              output | [----------] 100% pushing layers
              output | [          ]   0% pushing manifest for ghcr.io/elliotwutingfeng/tokei_rs:latest@sha256:1341c938af74d8074829ba25289e1bc77b90b9f41f208d7ee8eaa76ef14ee892
              output | [----------] 100% pushing manifest for ghcr.io/elliotwutingfeng/tokei_rs:latest@sha256:1341c938af74d8074829ba25289e1bc77b90b9f41f208d7ee8eaa76ef14ee892

 Push Summary ⏫
————————————————————————————————————————————————————————————————————————————————


 Local Output Summary 🎁 (disabled)
```

File: src/main.rs (hunk 2 above)
Comment (elliotwutingfeng) [reply src/main.rs]: Alright then, I've used serde rename.

File: Earthfile (hunk 1 above)
Comment (XAMPPRocky) [reply Earthfile:1]: Ah I think we just need to add `--push` to either the CI script or earthfile

Comment (elliotwutingfeng) [reply Earthfile:1]: The CI script already has `--push` i.e. `earthly --ci --push +docker --repo 'ghcr.io/elliotwutingfeng/tokei_rs'`

Comment (elliotwutingfeng) [reply Earthfile]: I got the `--push` working by placing the **ARG** statements within each target, it appears to be a scope-related bug similar to https://github.com/earthly/earthly/issues/2887

https://github.com/elliotwutingfeng/tokei_rs/actions/runs/5500568688/jobs/10023563976

**Comment by XAMPPRocky** [comment]
Thank you for your PR!

**Pull request merged**